hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -988,20 +1001,37 @@ private Map<String, List<HoodieRecord<HoodieMetadataPayload>>> readNonUniqueReco
     }
 
     HoodieTimer readTimer = HoodieTimer.start();
+    // baseFileRecordsMap: secondary key -> List of records in secondary index base file (possible that it is not compacted)
     Map<String, List<HoodieRecord<HoodieMetadataPayload>>> baseFileRecordsMap =
         fetchBaseFileAllRecordsByKeys(reader, sortedKeys, true, partitionName);
+    // logRecordsMap: secondaryKey -> Map[recordKey -> HoodieRecord] for the given secondary key
```

Comment:
yeah, all these looks non standard that we have to do it just for secondary index.